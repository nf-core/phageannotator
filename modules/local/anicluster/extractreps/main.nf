process ANICLUSTER_EXTRACTREPS {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta), path(fasta), path(clusters)

    output:
    tuple val(meta), path("*_representatives.fna.gz")   , emit: representatives
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    extractreps.py \\
        --fasta $fasta \\
        --clusters $clusters \\
        --out ${prefix}_representatives.fna

    gzip ${prefix}_representatives.fna

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_representatives.fna
    gzip ${prefix}_representatives.fna

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
    END_VERSIONS
    """
}
