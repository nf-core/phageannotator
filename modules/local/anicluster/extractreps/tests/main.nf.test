nextflow_process {

    name "Test process: ANICLUSTER_EXTRACTREPS"
    script "../main.nf"
    process "ANICLUSTER_EXTRACTREPS"


    test("fasta.gz & clusters.tsv") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true ),
                    file(params.pipelines_testdata_base_path + '/modules/local/anicluster/extractreps/clusters.tsv', checkIfExists: true )
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz & clusters.tsv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true ),
                    file(params.pipelines_testdata_base_path + '/modules/local/anicluster/extractreps/clusters.tsv', checkIfExists: true )
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
