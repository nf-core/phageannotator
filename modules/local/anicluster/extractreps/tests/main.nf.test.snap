{
    "fasta.gz & clusters.tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_anicluster_representatives.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,15b724eac7b61bbed8de17f17eae0bf8"
                ],
                "representatives": [
                    [
                        {
                            "id": "test"
                        },
                        "test_anicluster_representatives.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,15b724eac7b61bbed8de17f17eae0bf8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:03:20.478160213"
    },
    "fasta.gz & clusters.tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_anicluster_representatives.fna.gz:md5,b4ae3ed3aa26b50c3e3d3277e94ac940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,15b724eac7b61bbed8de17f17eae0bf8"
                ],
                "representatives": [
                    [
                        {
                            "id": "test"
                        },
                        "test_anicluster_representatives.fna.gz:md5,b4ae3ed3aa26b50c3e3d3277e94ac940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,15b724eac7b61bbed8de17f17eae0bf8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:03:11.703715193"
    }
}
